src/NorthwindWebsite/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -0,0 +1,207 @@
+
+            return new ChallengeResult(provider, properties);
+        }
+
+        public async Task<IActionResult> OnGetCallbackAsync(string returnUrl = null, string remoteError = null)
+        {
+            returnUrl = returnUrl ?? Url.Content(UrlConstants.ReturnToHomepageUrl);
+            if (!string.IsNullOrWhiteSpace(remoteError))
+            {
+                ErrorMessage = $"Error from external provider: {remoteError}";
+
+                return RedirectToPage("./Login", new { ReturnUrl = returnUrl });
+            }
+            var externalLoginInfo = await _signInManager.GetExternalLoginInfoAsync();
+            if (externalLoginInfo == null)
+            {
+                ErrorMessage = "Error loading external login information.";
+
+                return RedirectToPage("./Login", new { ReturnUrl = returnUrl });
+            }
+
+            // Sign in the user with this external login provider if the user already has a login.
+            var result = await _signInManager.ExternalLoginSignInAsync(
+                externalLoginInfo.LoginProvider,
+                externalLoginInfo.ProviderKey,
+                isPersistent: false,
+                bypassTwoFactor: true);
+            if (result.Succeeded)
+            {
+                _logger.LogInformation(
+                    "{Name} logged in with {LoginProvider} provider.", 
+                    externalLoginInfo.Principal.Identity.Name, 
+                    externalLoginInfo.LoginProvider);
+                return LocalRedirect(returnUrl);
+            }
+            if (result.IsLockedOut)
+            {
+                return RedirectToPage("./Lockout");
+            }
+
+            // If the user does not have an account, then ask the user to create an account.
+            ReturnUrl = returnUrl;
+            ProviderDisplayName = externalLoginInfo.ProviderDisplayName;
+            if (externalLoginInfo.Principal.HasClaim(claim => claim.Type == ClaimTypes.Email))
+            {
+                Input = new InputModel
+                {
+                    Email = externalLoginInfo.Principal.FindFirstValue(ClaimTypes.Email)
+                };
+            }
+
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostConfirmationAsync(string returnUrl = null)
+        {
+            returnUrl = returnUrl ?? Url.Content(UrlConstants.ReturnToHomepageUrl);
+            // Get the information about the user from the external login provider
+            var externalLoginInfo = await _signInManager.GetExternalLoginInfoAsync();
+            if (externalLoginInfo == null)
+            {
+                ErrorMessage = "Error loading external login information during confirmation.";
+                return RedirectToPage("./Login", new { ReturnUrl = returnUrl });
+            }
+
+            if (ModelState.IsValid)
+            {
```

Comment:
Is it possible to avoid making long nested condition here as well?

Reply:
No, not really

Reply:
Ok, let's leave it in this way then